### ISSUE TITLE
Make the Seaquest submarine resurface

### ISSUE THREAD

**Issue by LBerth** [Caterpillar314/paac#5]
With the PAAC implementation, the agents training on Seaquest are not able to resurface and score above 2000. However, many A3C implementations are able to score above 50k (see https://github.com/ppwwyyxx/tensorpack/tree/342cdc70c9a1f41d51b4482d0d64a519266ea828/examples/A3C-Gym). We will try to use different exploration techniques and different network architectures to solve this problem. See branch ExplorationPolicy.

**Comment by LBerth** [comment]
At the moment, we have tried these methods, without success : 
 - Using an e-greedy policy to choose the actions (with values for epsilon : 0.1 and 0.05)
 - Changing the temperature of the Boltzmann softmax function to choose the actions (with values for the temperature : 2 and 3)
 - Using the NATURE network architecture
 - Changing the entropy regularization strength parameter (with values ranging from 0.005 to 0.04)
 - Using a Bayesian Dropout network for the forward pass 
 - Using the same network as https://github.com/ppwwyyxx/tensorpack/tree/342cdc70c9a1f41d51b4482d0d64a519266ea828/examples/A3C-Gym , with 4 convolutionnal layers and max pooling.

See this issue on the PAAC repository : https://github.com/Alfredvc/paac/issues/3 . 
We will try to change the entropy regularization strength parameter with bigger values and add some random actions at the beginning of each environment.

**Comment by LBerth** [comment]
![image](https://user-images.githubusercontent.com/19271957/30054704-39d7d0d4-922d-11e7-8fd1-29cce07d1af6.png)

Increasing the entropy regularization strength parameter doesn't seem to help : see graphs of the reward and length of episode ( yellow 1, purple 0.5, green 0.1, red 0.05). The episode length can't go above 2200 : the submarine doesn't resurface.

**Comment by LBerth** [comment]
Eventually, we solved this issue when we implemented FiGAR (see issue #8 ).
By choosing to make the same move up to 10 times, the agent could explore more possibilities. It was able to go to the surface randomly a lot of times and consequently it understood the advantage of resurfacing.

![image](https://user-images.githubusercontent.com/19271957/33318085-a92015fc-d439-11e7-84b9-76379784a4da.png)
(Blue Figar 10 repetitions , Orange Paac default)

![resurfacing](https://user-images.githubusercontent.com/19271957/33316248-98a607c8-d433-11e7-820b-bdc3c6033c16.gif)
